core/src/main/java/org/apache/carbondata/core/datastore/DataRefNode.java

```diff
@@ -141,5 +141,5 @@ DimensionRawColumnChunk getDimensionChunk(FileHolder fileReader, int blockIndexe
    * @param pageNumber
    * @return
    */
-  int getPageRowCount(int pageNumber);
+  int getPageRowCount(int pageNumber) throws IOException;
```

Comment:
Avoid throwing IOException from this interface methods. Instead catch the IO exception in the caller methods and wrap it in RuntimeException and throw